dds/DCPS/security/AccessControl/XmlUtils.cpp

```diff
@@ -112,48 +114,268 @@ bool parse_bool(const XMLCh* in, bool& value)
   return true;
 }
 
+#define XMLDATETIME_HAS_GETEPOCH _XERCES_VERSION >= 30200
+#if !XMLDATETIME_HAS_GETEPOCH
+namespace {
+  bool parse_time_string(
+    const std::string& whole, size_t& pos, std::string& value, size_t width = std::string::npos)
+  {
+    const bool to_end = width == std::string::npos;
+    if (pos >= whole.size() || (!to_end && pos + width > whole.size())) {
+      return false;
+    }
+    try {
+      value = whole.substr(pos, width);
+    } catch (const std::out_of_range& e) {
+      return false;
+    }
+    if (!to_end) {
+      pos += width;
+    }
+    return true;
+  }
+
+  bool parse_time_field(
+    const std::string& whole, size_t& pos, int& value, size_t width = 2, int offset = 0)
```

Comment:
`offset` isn't adding much value, consider letting the caller handle it since it's not required for parsing